[refactor] Add ObserveHelper and remove some common function parameters

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26696 [refactor] move more functions to InsertObserversHelper
* #26642 [quant][graphmode] Skip observing bias across function call hierarchy
* **#26641 [refactor] Add ObserveHelper and remove some common function parameters**
* #26640 [quant][graphmode][fix] remove unneeded code

Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_observers*'
Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D17538668](https://our.internmc.facebook.com/intern/diff/D17538668)

jerryzh168 committed:
[refactor] Add ObserveHelper and remove some common function parameters

Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_observers*'
Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -172,16 +173,32 @@ Node* createIntReprNode(Value* v, Graph* g) {
+ public:
+  ObserveHelper(const ModuleQConfigMap& map) : module_qconfig_map_(map) {}
+  void insertObservers(script::Module& module, const std::string& method_name);
+  Node* insertObserverFor(
```
This and the next method don't need to be public.

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -172,16 +173,32 @@ Node* createIntReprNode(Value* v, Graph* g) {
   return intrepr;
 }
 
+class ObserveHelper {
```
Bikeshedding: maybe `InsertObserversHelper` to use the same pair of words everywhere?

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -172,16 +173,32 @@ Node* createIntReprNode(Value* v, Graph* g) {
+
+ private:
+  const ModuleQConfigMap& module_qconfig_map_;
+  std::unordered_set<Value*> values_to_skip_;
```
Maybe add some comments? Even though it is an internal class, the comments would still be helpful for people who would later try to understand the code.

ZolotukhinM reviewed (approved):
Looks good, thanks!

jerryzh168 committed:
Update on "[refactor] Add ObserveHelper and remove some common function parameters"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_observers*'
Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
